devdoc/CodeSubmissions.md

```diff
@@ -0,0 +1,76 @@
+* Prefer small submissions\
+  It isn't always possible, but several smaller PRs are much easier to review than one large change.  If your submission includes semi-automatic/mechanical changes (e.g. renaming large numbers of function calls, or adding an extra parameter) please keep it as a separate commit.  This makes it much easier to review the PR - since the reviewer will be looking for different errors in the different types of changes.
+* Check for silly mistakes\
+  Review your own code in github, after creating the PR to check for silly mistakes.  It doesn't take long, and often catches trivial issues.  It may avoid the need for a cycle of code-review/fixes.  It may be helpful to add some notes to specific changes e.g. "this change is mainly or solely refactoring method A into method B and C. ".  Some common examples of trivial issues to look for include:
+  - Inconsistent indentation, or using tabs rather than spaces to indent.
+  - Lines of tracing left in.
+  - Lines of code commented out that should be deleted.
+  - TBD reminders of work that need implementing or removing.
+  - Unrelated files that have been accidentally modified.
+  - Accidental changes to submodule versions.
+  - Typos in error messages, tracing or comments, or in the commit message.
+  - Incomplete edits when copy and pasting code.
+  - Check subtractions are the right way around, and potential for overflow.
+  - New files with the wrong copyright date
+* Check the target branch (see below)
+* Request one or more reviews.  For relatively simple changes a single reviewer is normally enough.
+
+## Reviewers
+
+All pull requests should be reviewed by someone who is not the author before merging.  Complex changes, changes that require input from multiple experts, or that have implications throughout the system should be reviewed by multiple reviewers.  This should include someone who is responsible for merging changes for that part of the system.  (Unless it is a simple change written by someone with review rights.)
+
+Contributors should use the github reviewers section on the PR to request reviews.  After a contributor has pushed a set of changes in response to a review, they should refresh the github review status, so the users are notified it is ready for re-review.  When the review is complete, a person with responsibility for merging changes to that part of the system should be added as a reviewer (or refreshed), with a comment that it is ready to merge.
+
+Reviewers should check for PRs that are ready for their review via github's webpage (filter "review-requested:\<reviewer-id>") or via the github CLI (e.g. gh pr status).  Contributors should similarly ensure they stay up to date with any comments on requests for change on their submissions.
+
```

Comment:
+ Reviewers should make an try to find all issues during the first review.  Often there are issues found by a reviewer, fixed by the contributor and then additional issues are reported by the reviewer (unrelated to the first set of changes)
+
+ If PR requires multiple reviewers, the contributor should communicate with reviewers for the parts that need multiple people to review.  This can help reduce the times where a reviewer makes comments, the contributor addresses that issue and a different reviewer objects to the change.  It would be better if the contributor can request all reviewers to comment before addressing the issue
+
+When doing a code review, the reviewer should be careful not to request additional features that are outside the scope of the initial jira (i.e. this feature would be better if you add these 2 new bits of functionality).  This should be a new jira request and not hold up the current PR